### DISCUSSION TITLE
Work queue cancellation

### DISCUSSION THREAD

**Discussion by ealeykin** [nats-io/nats-server#3669] [Q&A] [answered]
Hi,

I'm impressed with the way how NATS handles queues, however have a question about work queue items cancellation.

Input data is the following: `source` queue (stream) and `event` queue (stream). `Source` worker produces multiple events into `event` queue. I want to cancel entire source processing with it's sub-events.

Example:

`sources` stream - work queue with subjects `source.<id>`
`events` stream - work queue with subjects `event.<id>`

And cancellation as follows:

stream purge `sources` with subject `*.ID`
stream purge `events` with subject `*.ID`

What do you say about this approach or perhaps you would advise a better way to work out cancellation?

**Comment by derekcollison** [comment, marked as answer]
That looks reasonable. Assuming ID is repeated in the events stream, etc.
